src/libraries/Common/src/System/Security/Cryptography/MLDsa.cs

```diff
@@ -1207,87 +1482,245 @@ private AsnWriter ExportSubjectPublicKeyInfoCore()
                         writer.WriteObjectIdentifier(Algorithm.Oid);
                     }
 
-                    writer.WriteBitString(keySpan);
+                    writer.WriteBitString(publicKey);
                 }
 
+                Debug.Assert(writer.GetEncodedLength() <= capacity);
                 return writer;
             }
             finally
             {
-                // Public key doesn't need to be cleared
-                CryptoPool.Return(rented, 0);
+                // Public key does not need to be cleared.
+                CryptoPool.Return(rented, clearSize: 0);
             }
         }
 
         private AsnWriter ExportEncryptedPkcs8PrivateKeyCore(ReadOnlySpan<byte> passwordBytes, PbeParameters pbeParameters)
         {
-            ThrowIfDisposed();
-
-            // TODO: Determine a more appropriate maximum size once the format is actually known.
-            int size = Algorithm.SecretKeySizeInBytes * 2;
-            // The buffer is only being passed out as a span, so the derived type can't meaningfully
-            // hold on to it without being malicious.
-            byte[] rented = CryptoPool.Rent(size);
-            int written;
-
-            while (!TryExportPkcs8PrivateKey(rented, out written))
+            AsnWriter tmp = ExportPkcs8PrivateKeyCallback(static pkcs8 =>
             {
-                size = rented.Length;
-                CryptoPool.Return(rented, 0);
-                rented = CryptoPool.Rent(size * 2);
-            }
+                AsnWriter writer = new(AsnEncodingRules.BER, initialCapacity: pkcs8.Length);
```

Comment:
While this can be BER, we usually always encode in DER (a strict subset of BER), since it's not unheard of for other libraries to assume DER restrictions when reading.

Reply:
Oh.  WriteEncodedValue.  On a thing that could be lax-BER.

OK, that's the other reason to pass BER.  Never mind.